docs/SelectInput.md

```diff
@@ -0,0 +1,279 @@
+                />
+            </SimpleForm>
+        </Create>
+    );
+}
+```
+{% endraw %}
+
+Use the `create` prop when you want a more polished or complex UI. For example a Material UI `<Dialog>` asking for multiple fields because the choices are from a referenced resource.
+
+{% raw %}
+```js
+import {
+    SelectInput,
+    Create,
+    ReferenceInput,
+    SimpleForm,
+    TextInput,
+    useCreateSuggestionContext
+} from 'react-admin';
+
+import {
+    Box,
+    BoxProps,
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    TextField,
+} from '@material-ui/core';
+
+const PostCreate = () => {
+    return (
+        <Create>
+            <SimpleForm>
+                <TextInput source="title" />
+                <ReferenceInput source="category_id" reference="categories">
+                    <SelectInput create={<CreateCategory />} />
+                </ReferenceInput>
+            </SimpleForm>
+        </Create>
+    );
+}
+
+const CreateCategory = () => {
+    const { filter, onCancel, onCreate } = useCreateSuggestionContext();
+    const [value, setValue] = React.useState(filter || '');
+    const [create] = useCreate('categories');
+
+    const handleSubmit = (event) => {
+        event.preventDefault();
+        create(
+            {
+                payload: {
+                    data: {
+                        title: value,
+                    },
+                },
+            },
+            {
+                onSuccess: ({ data }) => {
+                    setValue('');
+                    onCreate(data);
+                },
+            }
+        );
+    };
+
+    return (
+        <Dialog open onClose={onCancel}>
+            <form onSubmit={handleSubmit}>
+                <DialogContent>
+                    <TextField
+                        label="New category name"
+                        value={value}
+                        onChange={event => setValue(event.target.value)}
+                        autoFocus
+                    />
+                </DialogContent>
+                <DialogActions>
+                    <Button type="submit">Save</Button>
+                    <Button onClick={onCancel}>Cancel</Button>
+                </DialogActions>
+            </form>
+        </Dialog>
+    );
+};
+```
+{% endraw %}
+
+## `sx`: CSS API
+
+The `<SelectInput>` component accepts the usual `className` prop. You can also override many styles of the inner components thanks to the `sx` property (as most Material UI components, see their [documentation about it](https://mui.com/customization/how-to-customize/#overriding-nested-component-styles)). This property accepts the following subclasses:
+
+| Rule name                | Description                                               |
+|--------------------------|-----------------------------------------------------------|
+| `& .RaSelectInput-input` | Applied to the underlying `ResettableTextField` component |
```

Comment:
I don't fully understand the `SelectInput` code on this. But is clear that is not the same

Reply:
Should it be `RaSelectInput-inputEnd` in the docs reference? It was just `input` before

Reply:
Ok